extension/encoding/awslogsencodingextension/unmarshaler/subscription-filter/subscription_filter.go

```diff
@@ -0,0 +1,140 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package subscriptionfilter // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/awslogsencodingextension/unmarshaler/subscription-filter"
+
+import (
+	"bytes"
+	"compress/gzip"
+	"errors"
+	"fmt"
+	"io"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-lambda-go/events"
+	jsoniter "github.com/json-iterator/go"
```

Comment:
jsoniter is unmaintained since 2021, prefer goccy-json.

Reply:
Perfect, thanks!!